src/examples/libpmemobj/libart/libart2/art.c

```diff
@@ -0,0 +1,1654 @@
+/*
+ * Copyright 2017, FUJITSU TECHNOLOGY SOLUTIONS GMBH
+ * Copyright 2012, Armon Dadgar. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in
+ *       the documentation and/or other materials provided with the
+ *       distribution.
+ *
+ *     * Neither the name of the copyright holder nor the names of its
+ *       contributors may be used to endorse or promote products derived
+ *       from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/*
+ * ===========================================================================
+ *
+ *       Filename:  art.c
+ *
+ *    Description:  implement ART tree using libpmemobj based on libart
+ *
+ *         Author:  Andreas Bluemle, Dieter Kasper
+ *                  andreas.bluemle@itxperts.de
+ *                  dieter.kasper@ts.fujitsu.com
+ *
+ *   Organization:  FUJITSU TECHNOLOGY SOLUTIONS GMBH
+ * ============================================================================
+ */
+
+/*
+ * based on https://github.com/armon/libart/src/art.c
+ */
+
+#include <assert.h>
+#include <errno.h>
+#include <unistd.h>
+#include <string.h>
+#include <strings.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <limits.h>
+#include <stdbool.h>
+#include <fcntl.h>
+#include <emmintrin.h>
+#include <sys/types.h>
+#include "libpmemobj.h"
+#include "art.h"
+
+PMEMoid null_oid;
+
+size_t art_tree_root_type_num;
+size_t art_node4_type_num;
+size_t art_node16_type_num;
+size_t art_node48_type_num;
+size_t art_node256_type_num;
+size_t art_leaf_type_num;
+
+int art_tree_init(PMEMobjpool *pop, int *newpool);
```

Comment:
`make_leaf` is not used outside `libart2/art.c` -- it should be defined as `static PMEMoid make_leaf`.

It is already defined before all cases where it is called in `art.c`. There is no need for a separate declaration here.
BTW, the compiler only complains about missing prototype in case of symbols with external linkage -- this is useful to warn the programmer about "you forgot to add a declaration to a header file!"
This routine is not used anywhere else, so I assume it should have internal linkage.